tests/python/test_gc.py

```diff
@@ -79,3 +94,38 @@ def test_pointer_gc():
 
         # Note that being inactive doesn't mean it's not allocated.
         assert L.num_dynamically_allocated == 1
+
+
+@ti.test(require=[ti.extension.sparse, ti.extension.async_mode],
+         async_mode=True)
+def test_fuse_allocator_state():
+    N = 16
+    x = ti.field(dtype=ti.i32, shape=N)
+    y = ti.field(dtype=ti.i32)
+
+    y_parent = ti.root.pointer(ti.i, N * 2)
+    y_parent.place(y)
+
+    # https://github.com/taichi-dev/taichi/pull/1973#pullrequestreview-511154376
+
+    @ti.kernel
+    def activate_y():
+        for i in x:
+            idx = i + 1
+            y[idx] = idx
+
+    @ti.kernel
+    def deactivate_y():
+        for i in x:
+            ti.deactivate(y_parent, i)
+
+    activate_y()
+    deactivate_y()
+    ti.sync()
+
+    # TODO: assert that activate_y and deactivate_y are not fused.
```

Comment:
I've confirmed this with the dumped dot graph. But task fusion is currently not recorded in the stats yet.